python/guide/packages/gluon/data.md

```diff
@@ -1,325 +1,120 @@
-The following examples add the label by 10 besides resizing an image:
+When we retrieve the same sample as before from the dataset, we now see an augmented version of the image (with the same label). We'd see a different augmented image every time we retrieve this sample.
 
-```{.python .input}
-dogcat_both = dogcat.transform(lambda x, y: (resize(x), y+10))
-dogcat_both[0][1]
 ```
-
-The `vision.transform` submodule provide multiple pre-defined data
-transformation methods. For example, the following example chains the image
-resize and `ToTensor`, which transforms the data layout into (C x H x W) with
-float32 data type. Please refer to [Image Augmentation](./image-augmentation.md)
-for more details.
-
-```{.python .input  n=3}
-transforms = gdata.vision.transforms.Compose([
-    gdata.vision.transforms.Resize((24, 24)),
-    gdata.vision.transforms.ToTensor()])
-mnist_transformed = mnist_train.transform_first(transforms, lazy=True)
-(mnist_train[0][0].shape, '->', mnist_transformed[0][0].shape)
+data_sample, label_sample = dataset[42]
+print('Clothing Category #{}'.format(label_sample))
+plot(data_sample)
 ```
 
-## Sample Mini-batches 
-
-If we train a neural network with mini-batch SGD, we
-need to sample a mini-batch for every iteration. Class `DataLoader` samples a
-dataset into mini-batches. In the following example, we create a `DataLoader`
-instance, which is an iterator that returns a mini-batch each time.
+## Getting started with [`DataLoader`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.DataLoader.html)s
 
-```{.python .input  n=18}
-data = gdata.DataLoader(dataset, batch_size=4)
-for X, y in data:
-    print('X shape:', X.shape, '\ty:', y.asnumpy())
-```
-
-Since the number of examples can not be divided by the batch size, the last min-
-batch only has two examples. We can chose to ignore the last incomplete mini-
-batch:
+Our `dataset` gives us individual samples, but we usually train neural networks on batches of samples. A [`DataLoader`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.DataLoader.html) retrieves samples from the `dataset` and stacks them into batches. At a minimum, all we need to specify is the number of samples we want in each batch, called the `batch_size`, but [`DataLoader`](https://beta.mxnet.io/api/gluon/_autogen/mxnet.gluon.data.DataLoader.html)s have many other useful features. Shuffling data samples for training is as simple as setting `shuffle`.
 
-```{.python .input  n=19}
-data = gdata.DataLoader(dataset, batch_size=4, last_batch='discard')
-for X, y in data:
-    print('y:', y.asnumpy())
 ```
-
-Or put it into the beginning of the next epoch:
-
-```{.python .input  n=23}
-data = gdata.DataLoader(dataset, batch_size=4, last_batch='rollover')
-for X, y in data:
-    print('epoch 0, y:', y.asnumpy())
-for X, y in data:
-    print('epoch 1, y:', y.asnumpy())
+dataloader = mx.gluon.data.DataLoader(dataset,
+                                      batch_size=12345,
```

Comment:
Let's try giving meaningful numbers, like 100, in case it would fail with OOM on a small machine.

Reply:
Was trying to avoid too many batches showing. Will add a comment about this though.